panther_hardware_interfaces/src/gpio_driver.cpp

```diff
@@ -5,56 +5,282 @@ namespace panther_hardware_interfaces
 
 GPIOController::GPIOController()
 {
-  chip_ = std::make_unique<gpiod::chip>("gpiochip0");
-  vmot_ =
-    std::make_unique<gpiod::line>(chip_->get_line(static_cast<unsigned int>(GPIOPins::VMOT_ON)));
-  motor_driver_ = std::make_unique<gpiod::line>(
-    chip_->get_line(static_cast<unsigned int>(GPIOPins::MOTOR_DRIVER_EN)));
-  e_stop_reset_ = std::make_unique<gpiod::line>(
-    chip_->get_line(static_cast<unsigned int>(GPIOPins::E_STOP_RESET)));
-  watchdog_ =
-    std::make_unique<gpiod::line>(chip_->get_line(static_cast<unsigned int>(GPIOPins::WATCHDOG)));
+  auto gpio_chip = gpiod::chip(gpio_chip_path_);
+
+  std::vector<GPIOpins> controll_pins;
+  for (GPIOpins pin = GPIOpins::WATCHDOG + 1; pin < GPIOpins::UNKNOWN; pin = pin + 1) {
+    controll_pins.push_back(pin);
+  }
+
+  controll_request_ = create_line_request(gpio_chip, controll_pins);
+  watchdog_request_ = create_line_request(gpio_chip, GPIOpins::WATCHDOG);
+
+  gpio_monitor_on();
```

Comment:
I think that it would make sense to split it into initialize and activate (activate - `gpio_monitor_on`)